src/index.tsx

```diff
@@ -40,9 +40,30 @@ const RHFInput = React.memo(
     const [inputValue, setInputValue] = React.useState(defaultData);
     const valueRef = React.useRef(defaultData);
     const methods = useFormContext();
-    const setValue = methods ? methods.setValue : setValueFromProp;
-    const register = methods ? methods.register : registerFromProp;
-    const unregister = methods ? methods.unregister : unregisterFromProp;
+    const setValue =
+      setValueFromProp !== undefined
```

Comment:
I think we can make it simpler to  `setValueFromProp ?  methods.setValue :  setValueFromProp`. But you know what lovely work I will approve it anyhow.